syntax = "proto3";
package rag;

import "rag_chunks.proto";

enum ScopeType {
    UNKNOWN_SCOPE = 0;
    ASSIGNMENT_SCOPE = 1;
    COURSE_SCOPE = 2;
    ORG_SCOPE = 3;
    ALL_SCOPE = 4;
}

message DocumentSource {
    string public_uuid = 1;
    string version_uuid = 2;
    string name = 3;
    string scope_uuid = 4;
    ScopeType scope_type = 5;
    uint64 version_timestamp = 8;
}

enum FileType {
    UNKNOWN = 0;
    MARKDOWN = 1;
    TEXT = 2;
    PYTHON = 3;
    CODE = 4;
}

message SupportingDocument {
    string public_uuid = 1;
    string version_uuid = 2;
    string source_public_uuid = 3;
    string name = 4;
    string content = 5;
    FileType file_type = 6;
    uint64 version_timestamp = 7;
}

enum DocumentType {
    UNKOWN_CONTENT = 0;
    SUPPORTING = 1;
    ASSIGNMENT = 2;
    SUBMISSION = 3;
}

message DocumentChunks {
    string document_version_uuid = 1;
    string rag_method_info = 2;
    DocumentType document_type = 3;
    rag_chunks.DocumentRoot document_root = 4;
    repeated rag_chunks.Chunk chunks = 5;
    uint64 end_timestamp = 6;
}


message ChunkSummary {
    string chunk_uuid = 1;
    string version_uuid = 2;
    string summary = 3;
}

message DocumentChunkSummaries {
    string document_version_uuid = 1;
    string summarization_method_info = 2;
    repeated ChunkSummary chunk_summaries = 3;
    uint64 end_timestamp = 4;
}


message Embedding {
    repeated float embedding = 1;
}

message PassageEmbedding {
    string passage_uuid = 1;
    repeated Embedding embeddings = 3;
}

message DocumentPassageEmbeddings {
    string document_version_uuid = 1;
    string embedding_method_info = 2;
    repeated PassageEmbedding passage_embeddings = 3;
    uint64 end_timestamp = 4;
}